File: packages/ts/src/modules/platform-keys.ts
import type { HubflyClient } from '../client.js';
import type { ApiResponse, SubaccountKey } from '../types.js';

export class PlatformKeysModule {
  constructor(private readonly client: HubflyClient) {}

  public async list(parentType: 'user' | 'organization', parentId?: string): Promise<ApiResponse<SubaccountKey[]>> {
    return this.client.request<SubaccountKey[]>('/api/v1/platform/api-keys', { query: { parentType, parentId } });
  }

  public async create(params: { parentType: 'user' | 'organization'; parentId?: string; name?: string; scopes?: string[]; expiresAt?: string | null }): Promise<ApiResponse<SubaccountKey>> {
    return this.client.request<SubaccountKey>('/api/v1/platform/api-keys/create', { method: 'POST', body: params });
  }

  public async revoke(keyId: string): Promise<ApiResponse<{ ok: boolean }>> {
    return this.client.request<{ ok: boolean }>(`/api/v1/platform/api-keys/${keyId}/revoke`, { method: 'POST' });
  }
}

File: packages/go/platform_keys.go
package hubfly

import (
	"context"
	"fmt"
)

type PlatformKeysService struct{ client *Client }

func (s *PlatformKeysService) List(ctx context.Context, parentType, parentID string) (*Response[[]SubaccountKey], error) {
	return Request[[]SubaccountKey](ctx, s.client, "GET", fmt.Sprintf("/api/v1/platform/api-keys?parentType=%s&parentId=%s", parentType, parentID), nil)
}

func (s *PlatformKeysService) Create(ctx context.Context, params map[string]interface{}) (*Response[SubaccountKey], error) {
	return Request[SubaccountKey](ctx, s.client, "POST", "/api/v1/platform/api-keys/create", params)
}

func (s *PlatformKeysService) Revoke(ctx context.Context, keyID string) (*Response[map[string]bool], error) {
	return Request[map[string]bool](ctx, s.client, "POST", fmt.Sprintf("/api/v1/platform/api-keys/%s/revoke", keyID), nil)
}
